src/components/elements/Inputs/FileInput/FilePreviewCard.tsx

```diff
@@ -70,36 +111,61 @@ const FilePreviewCard = ({ file, className, showPrivateCheckbox, onDelete, onPri
 interface UploadingProps {
   title: string;
   file: UploadedFile;
+  variant: FilePreviewCardVariant;
 }
 
-const Uploading = ({ title, file }: UploadingProps) => {
+const Uploading = ({ title, file, variant }: UploadingProps) => {
   const t = useT();
   const { format } = useFileSize();
+  const subtitleMap = {
+    image: "Image is being uploaded.",
+    geoFile: "Data is being uploaded."
+  };
 
+  let subtitle;
+  if (variant.type !== undefined && subtitleMap[variant.type] !== undefined) {
```

Comment:
This can be simplified to `if (subtitleMap[variant?.type] !== undefined) {`

Reply:
im gettin that error when variant.type !== undefined is deleted
The error Type 'undefined' cannot be used as an index type is caused by the fact that variant?.type can be undefined, and using undefined as a key in an object is not allowed.